lib/db/set.go

```diff
@@ -124,6 +124,10 @@ func (s *FileSet) Drop(device protocol.DeviceID) {
 
 func (s *FileSet) Update(device protocol.DeviceID, fs []protocol.FileInfo) {
 	l.Debugf("%s Update(%v, [%d])", s.folder, device, len(fs))
+
+	// do not modify fs in place, it is still used in outer scope
+	fs = append([]protocol.FileInfo{}, fs...)
```

Comment:
This can just be fs := fs no?

Reply:
I just tested it (with  something like `newFs := fs` due to "no new variable") and that still uses the same backing array. I assumed this was the case before due to this [official wiki entry on slice copying](https://github.com/golang/go/wiki/SliceTricks#copy).

Reply:
I think you can just do `append(nil, fs...)` in that case.

Reply:
Go apparently still wants a typed nil: `append([]protocol.FileInfo(nil), fs...)`